### ISSUE TITLE
EXC_BAD_ACCESS on if (otherString->unicodeCharacters() == NULL)

### ISSUE THREAD

**Issue by vu0tran** [MailCore/mailcore2#1073]
In the method
````
int String::compareWithCaseSensitive(String * otherString, bool caseSensitive)
`````
Sometimes my app crashes when trying to access "otherString->unicodeCharacters()" which is NULL.

**Comment by dinhviethoa** [comment]
Here's an excerpt of String::compareWithCaseSensitive() implementation.
Could you let me know where it does fall through? is otherString NULL ?

Could you also show the stack trace of the crash?

```
int String::compareWithCaseSensitive(String * otherString, bool caseSensitive)
{
    if ((length() == 0) && (otherString->length() == 0)) {
        return 0;
    }
    if ((unicodeCharacters() == NULL) && (otherString->unicodeCharacters() != NULL)) {
        return 0;
    }
    
    if (unicodeCharacters() == NULL) {
        return -1;
    }
    
    if (otherString->unicodeCharacters() == NULL) {
        return -1;
    }
```

**Comment by vu0tran** [comment]
otherString in the debugger is NULL and fails at: if (otherString->unicodeCharacters() == NULL)

When trying to access unicodeCharacters() on NULL object. 

I'm trying to repro it to provide the stack trace but can't do it again, but it pops up every once in a while. Will update here when I got the stack trace.

**Comment by dinhviethoa** [comment]
Fixed with 1714bc64ddb5d9646eb21c6eb6999933f37ffbfe.
